connectors/src/connectors/slack/temporal/activities.ts

```diff
@@ -0,0 +1,339 @@
+
+  const c: ConversationsHistoryResponse = await client.conversations.history({
+    channel: channelId,
+    limit: 300,
+    oldest: `${startTsSec}`,
+    latest: `${endTsSec}`,
+    cursor: nextCursor,
+  });
+  if (c.error) {
+    throw new Error(
+      `Failed getting messages for channel ${channelId}: ${c.error}`
+    );
+  }
+  if (c.messages === undefined) {
+    throw new Error(
+      `Failed getting messages for channel ${channelId}: messages is undefined`
+    );
+  }
+  for (const message of c.messages) {
+    messages.push(message);
+  }
+  const text = await formatNonThreaded(channelId, messages);
+
+  const startDate = new Date(startTsMs);
+  const endDate = new Date(endTsMs);
+  const startDateStr = `${startDate.getFullYear()}-${startDate.getMonth()}-${startDate.getDate()}`;
+  const endDateStr = `${endDate.getFullYear()}-${endDate.getMonth()}-${endDate.getDate()}`;
+  const documentId = `${channelName}-nonthreaded-${startDateStr}-${endDateStr}`;
+  await upsertToDatasource(dataSourceConfig, documentId, text);
+
+  return c;
+}
+
+export async function syncThreads(
+  dataSourceConfig: DataSourceConfig,
+  slackAccessToken: string,
+  channelId: string,
+  threadsTs: string[]
+) {
+  while (threadsTs.length > 0) {
+    const _threadsTs = threadsTs.splice(0, MAX_CONCURRENCY_LEVEL);
+
+    await Promise.all(
+      _threadsTs.map((t) =>
+        syncThread(dataSourceConfig, slackAccessToken, channelId, t)
+      )
+    );
+  }
+}
+
+export async function syncThread(
+  dataSourceConfig: DataSourceConfig,
+  slackAccessToken: string,
+  channelId: string,
+  threadTs: string
+) {
+  const client = new WebClient(slackAccessToken);
+
+  let allMessages: Message[] = [];
+  const { channel } = await client.conversations.info({
+    channel: channelId,
+  });
+  if (!channel) {
+    throw new Error("Channel not found for id " + channelId);
+  }
+
+  let next_cursor = undefined;
+
+  do {
+    const replies: ConversationsRepliesResponse =
+      await client.conversations.replies({
+        channel: channelId,
+        ts: threadTs,
+        cursor: next_cursor,
+      });
+    if (replies.error) {
+      throw new Error(replies.error);
+    }
+    if (!replies.messages) {
+      break;
+    }
+    allMessages = allMessages.concat(replies.messages);
+    next_cursor = replies.response_metadata?.next_cursor;
+  } while (next_cursor);
+
+  const text = await formatThread(channelId, allMessages);
+  const documentId = `${channel.name}-threaded-${threadTs}`;
+
+  await upsertToDatasource(dataSourceConfig, documentId, text);
+}
+
+async function formatThread(channelId: string, messages: Message[]) {
+  const promises = messages.map(async (message) => {
+    const userName = await getUserName(message.user as string);
+    let text = await processMessageForMentions(message.text as string);
+    text = processMessageRemoveEmoji(text);
+
+    return `${userName} said:\n ${text}`;
+  });
+
+  return (await Promise.all(promises)).join("\n");
```

Comment:
`\n` is a bit weak as a separator along with the `xxx said:` I think.

Let's go for a slightly more structured format:

```
>> @spolu [20230427 14:32]
{message content}
>> @lasryaric [20230427 14:33]
{message content}
```

I think we want to have that logic in a separate function so that it's correctly shared across threaded/non-threaded